CommonCompilerConfig.cmake

```diff
@@ -1,408 +1,417 @@
+#
+# EXTERNAL_LINKER_FLAGS_UNOPTIMIZED: string of linker flags for linking 3rd-party executables
+#                      and shared libraries (DLLs) without explicit optimization
+#                      settings. This property is for use with
+#                      EXTERNAL_C_FLAGS_UNOPTIMIZED and EXTERNAL_CXX_FLAGS_UNOPTIMIZED
+#
+# EXTERNAL_LINKER_FLAGS_OPTIMIZED: string of linker flags for linking 3rd-party executables
+#                      and shared libraries (DLLs) with optimizations that are
+#                      compatible with EXTERNAL_C_FLAGS_OPTIMIZED and EXTERNAL_CXX_FLAGS_OPTIMIZED
+#
+# EXTERNAL_STATICLIB_CMAKE_DEFINITIONS_OPTIMIZED: list of -D cmake definitions corresponding to
+#                      EXTERNAL_C_FLAGS_OPTIMIZED (e. g. use of gcc-ar and gcc-ranlib wrappers for gcc >= 4.9
+#                      in combination with Link Time Optimization)
+#
+# EXTERNAL_STATICLIB_CONFIGURE_DEFINITIONS_OPTIMIZED: variant of
+#                      EXTERNAL_STATICLIB_CMAKE_DEFINITIONS_OPTIMIZED used for
+#                      configure-based builds
+#
+# INTERNAL_CXX_FLAGS_OPTIMIZED: string of C++ flags with explicit optimization flags for internal sources
+#
+# INTERNAL_LINKER_FLAGS_OPTIMIZED: string of linker flags for linking internal executables
+#                      and shared libraries (DLLs) with optimizations that are
+#                      compatible with INTERNAL_CXX_FLAGS_OPTIMIZED
+#
+# PYEXT_LINKER_FLAGS_OPTIMIZED: string of linker flags for linking python extension
+#                      shared libraries (DLLs) with optimizations that are
+#                      compatible with EXTERNAL_CXX_FLAGS_OPTIMIZED.
+#
+# CMAKE_AR: Name of archiving tool (ar) for static libraries. See cmake documentation
+#
+# CMAKE_RANLIB: Name of randomizing tool (ranlib) for static libraries. See cmake documentation
+#
+# CMAKE_LINKER: updated, if needed; use ld.gold if available. See cmake
+#               documentation
+#
+# NOTE The XXX_OPTIMIZED flags are quite aggresive - if your code misbehaves for
+# strange reasons, try compiling without them.
+
+# NOTE much of the code below was factored out from src/CMakeLists.txt
+
+if(NOT DEFINED PLATFORM)
+    message(FATAL_ERROR "PLATFORM property not defined: PLATFORM=${PLATFORM}")
+endif()
+
+include(CheckCXXCompilerFlag)
+
+
+# Init exported properties
+set(COMMON_COMPILER_DEFINITIONS)
+set(COMMON_COMPILER_DEFINITIONS_STR)
+
+set(INTERNAL_CXX_FLAGS_OPTIMIZED)
+set(INTERNAL_LINKER_FLAGS_OPTIMIZED)
+
+set(PYEXT_LINKER_FLAGS_OPTIMIZED)
+
+set(EXTERNAL_C_FLAGS_UNOPTIMIZED)
+set(EXTERNAL_C_FLAGS_OPTIMIZED)
+
+set(EXTERNAL_CXX_FLAGS_UNOPTIMIZED)
+set(EXTERNAL_CXX_FLAGS_OPTIMIZED)
+
+set(EXTERNAL_LINKER_FLAGS_UNOPTIMIZED)
+set(EXTERNAL_LINKER_FLAGS_OPTIMIZED)
+
+set(EXTERNAL_STATICLIB_CMAKE_DEFINITIONS_OPTIMIZED)
+set(EXTERNAL_STATICLIB_CONFIGURE_DEFINITIONS_OPTIMIZED)
+
+##### Set parameters
+
+set(INTERNAL_CPP_STANDARD "-std=c++11")

```

Comment:
I think this is just more confusing than stating it directly, unless we have a good usecase.